packages/content-utils/src/runtime/fancyContent.ts

```diff
@@ -0,0 +1,83 @@
+/// <reference types="astro/client" />
+
+import { defineCollection as defineNative, z, type SchemaContext } from 'astro:content';
+
+// Copied from Starlight because Astro core doesn't export their definitions :(
```

Comment:
Do you think it's worth opening a PR in astro for this?

Reply:
Maybe, I'll bring this up on Discord